core/metacling/test/TClingTests.cxx

```diff
@@ -0,0 +1,13 @@
+#include "TInterpreter.h"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+TEST(TCling, GenerateDictionary)
+{
+   // Make sure we do not crash or go in infinite loop.
+   ASSERT_TRUE(gInterpreter->GenerateDictionary("std::vector<int>"));
```

Comment:
vector<int> might not be the best of choice since it has a pre-built dictionary.  I would recommend one we are unlikely to pre-build (vector<TNamed*> for example).

I would also add test that the TClass exist and 'IsLoaded'. (and is not before).

Reply:
Could you give the snippet that you have in mind?

Reply:
```
auto cl = TClass::GetClass("vector<const TNamed*>");
if (cl) ASSERT_FALSE(cl->IsLoaded());
ASSERT_TRUE(gInterpreter->GenerateDictionary("std::vector<const TNamed*>"));
cl = TClass::GetClass("vector<const TNamed*>");
ASSERT_TRUE(cl != nullptr);
ASSERT_TRUE(cl->IsLoaded());
```
and for each call to GenerateDictionary I would use a different containee type.

Reply:
Using `const` in `vector<const TNamed*>` hits another bug... I am using just TNamed...